go/src/golangApi/db/db.go

```diff
@@ -0,0 +1,68 @@
+package db
+
+import (
+  "database/sql"
+  "fmt"
+  o "golangApi/models"
+  _ "github.com/lib/pq"
+)
+
+var db *sql.DB
+
+const (
+    dbhost = "localhost"
+    dbport = "5432"
+    dbuser = "t_schafer"
+    dbpass = "\"\""
+    dbname = "benchmark_compare_dev"
+)
+
+func InitDb() {
+  var err error
+  psqlInfo := fmt.Sprintf("host=%s port=%s user=%s "+
+        "password=%s dbname=%s sslmode=disable",
+        dbhost, dbport, dbuser, dbpass, dbname)
+
+  db, err = sql.Open("postgres", psqlInfo)
+  if err != nil {
+        panic(err)
+    }
+  err = db.Ping()
+  if err != nil {
+      panic(err)
+  }
+}
+
+func QueryOlympians(data *o.Olympians) error {
```

Comment:
I think I understand most of this function. You are goin through the response from the database row by row, creating and appending an OlympianSummary object for each row to the data argument (which is a list of Olympian summary objects). I don't actually see how the data is being returned here. Is it kind of like a javascript Promise, where you have to pull the response out of it later, like in the router or something?

Reply:
The data for this function is returned on line 61:
`data.Olympians = append(data.Olympians, olympian)`

Go is a little different, where you can use pointers to update objects in their exact spot in memory. So rather than having a 'copy' of Olympians that I am sending back, I am simply updating that one object. The updated Olympians array is then returned in the olympiansHandler in the main.go package.

Reply:
It's really hard to explain, I'm still pretty new to the language! We can jump on a slack call and go over the whole thing though. It's all very fascinating.

Reply:
That makes sense. Go seems very different and cool.